### PR TITLE
add account support check in validator

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation

Mirrors the [wallet_createSession handler
](https://github.com/MetaMask/metamask-extension/pull/27782/files#diff-107459889087f2776c6db636bd45498bef6749302f9d2dc633b4de17fede40a3R96-R108) in how eth account support is checked/asserted. Opted to do this rather than modify `assertScopeSupported` because the `bucketScopes` helper also relies on `assertScopedSupported` but doesn't care about accounts (which is why eth accounts are checked outside of assertScopeSupported in the wallet_createSession handler currently)

### `@metamask/package-a`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

### `@metamask/package-b`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [ ] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: packages/multichain/src/caip25Permission.ts [hunk 1]
```diff
@@ -120,6 +122,26 @@ const specificationBuilder: PermissionSpecificationBuilder<
         isChainIdSupported,
       });
 
+      // These should be EVM accounts already although the name does not necessary imply that
+      // These addresses are lowercased already
```
Comment (adonesky1) [inline packages/multichain/src/caip25Permission.ts:126]:
```suggestion
      // Fetch EVM accounts from native wallet keyring
      // These addresses are lowercased already
```

File: packages/multichain/src/caip25Permission.test.ts [hunk 2]
```diff
@@ -549,6 +552,61 @@ describe('endowment:caip25', () => {
       expect(isChainIdSupportedBody).toContain('findNetworkClientIdByChainId');
     });
 
+    it('throws if the eth accounts specified in the normalized scopeObjects are not all supported', () => {
```
Comment (adonesky1) [inline packages/multichain/src/caip25Permission.test.ts:555]:
```suggestion
    it('throws if the eth accounts specified in the normalized scopeObjects are not found in the wallet keyring', () => {
```

File: packages/multichain/src/caip25Permission.ts [hunk 3]
```diff
@@ -120,6 +122,26 @@ const specificationBuilder: PermissionSpecificationBuilder<
...
+        optionalScopes: normalizedOptionalScopes,
+        isMultichainOrigin,
+      }).map((address) => address.toLowerCase() as Hex);
+
+      const allEthAccountsSupported = ethAccounts.every((address) =>
+        existingEvmAddresses.includes(address),
+      );
+      if (!allEthAccountsSupported) {
+        throw new Error(
+          `${Caip25EndowmentPermissionName} error: Received invalid eip155 account values for caveat of type "${Caip25CaveatType}".`,
```
Comment (adonesky1) [inline packages/multichain/src/caip25Permission.ts:141]:
```suggestion
          `${Caip25EndowmentPermissionName} error: Received eip155 account value(s) for caveat of type "${Caip25CaveatType}" that were not found in the wallet keyring`,
```

File: packages/multichain/src/caip25Permission.test.ts [hunk 4]
```diff
@@ -493,7 +496,7 @@ describe('endowment:caip25', () => {
           },
         },
         normalizedOptionalScopes: {
-          'eip155:1': {
+          'eip155:5': {
```
Comment (adonesky1) [inline packages/multichain/src/caip25Permission.test.ts:499]: why?

Comment (jiexi) [reply packages/multichain/src/caip25Permission.test.ts:499]: to match the input that validator() is called with (eip155:5 is passed in the optionalScopes, not eip155:1). This is of course unrelated to the validator changes being made in this PR

**Review by adonesky1 — APPROVED** [review]
LGTM

**Pull request merged**